tests/deepsparse/transformers/pipelines/test_text_generation.py

```diff
@@ -425,6 +425,39 @@ def test_num_generated_predictions(self, setup):
         for generation in output_sequences.generations:
             assert len(generation) == 2
 
+    def test_token_generation__deterministic(self, setup):
+        """Should generate the same tokens"""
+        pipeline_kwargs = {
+            "task": "text_generation",
+            "model_path": self.model_stub,
+            "deterministic": True,
+        }
+        pipeline = self.get_pipeline(**pipeline_kwargs)
+        inference = pipeline(
+            sequences=["hello?"], num_generated_predictions=3, max_tokens=100
+        )
+        sequences = inference.sequences[0]
+        # Output should be the same from one another
+        assert all(sequence == sequences[0] for sequence in sequences)
+
+    def test_token_generation__non_deterministic(self, setup):
+        """Should generate non-unique tokens"""
+        pipeline_kwargs = {
+            "task": "text_generation",
+            "model_path": self.model_stub,
+            "deterministic": False,
+        }
+        pipeline = self.get_pipeline(**pipeline_kwargs)
+        num_generated_predictions = 3
+        inference = pipeline(
+            sequences=["hello?"],
+            num_generated_predictions=num_generated_predictions,
+            max_tokens=100,
+        )
+        sequences = inference.sequences[0]
+        # Outputs should be unique from one another
+        assert len(set(sequences)) == num_generated_predictions
```

Comment:
Could we add an assert to confirm they differ?

Reply:
I think if there are duplicated exact sequences the len(set(seqs)) of those will be 1, so this test should confirm that they differ